vita3k/modules/SceCommonDialog/SceCommonDialog.cpp

```diff
@@ -805,54 +805,52 @@ EXPORT(int, sceSaveDataDialogAbort) {
 }
 
 static void check_empty_param(EmuEnvState &emuenv, const SceAppUtilSaveDataSlotEmptyParam *empty_param, const uint32_t idx) {
-    vfs::FileBuffer thumbnail_buffer;
     emuenv.common_dialog.savedata.title[idx] = "";
     emuenv.common_dialog.savedata.subtitle[idx] = "";
     emuenv.common_dialog.savedata.icon_buffer[idx].clear();
+    emuenv.common_dialog.savedata.icon_texture[idx] = {};
     emuenv.common_dialog.savedata.has_date[idx] = false;
+    emuenv.common_dialog.savedata.slot_info[idx].isExist = 0;
     if (empty_param) {
         emuenv.common_dialog.savedata.title[idx] = empty_param->title.get(emuenv.mem) ? empty_param->title.get(emuenv.mem) : emuenv.common_dialog.lang.save_data.save["new_saved_data"];
         auto iconPath = empty_param->iconPath.get(emuenv.mem);
         SceUChar8 *iconBuf = reinterpret_cast<SceUChar8 *>(empty_param->iconBuf.get(emuenv.mem));
         auto iconBufSize = empty_param->iconBufSize;
+        vfs::FileBuffer thumbnail_buffer;
         if (iconPath) {
             auto device = device::get_device(empty_param->iconPath.get(emuenv.mem));
             auto thumbnail_path = translate_path(empty_param->iconPath.get(emuenv.mem), device, emuenv.io.device_paths);
             vfs::read_file(VitaIoDevice::ux0, thumbnail_buffer, emuenv.pref_path, thumbnail_path);
             emuenv.common_dialog.savedata.icon_buffer[idx] = thumbnail_buffer;
-            emuenv.common_dialog.savedata.icon_loaded[idx] = true;
         } else if (iconBuf && iconBufSize != 0) {
             thumbnail_buffer.insert(thumbnail_buffer.end(), iconBuf, iconBuf + iconBufSize);
             emuenv.common_dialog.savedata.icon_buffer[idx] = thumbnail_buffer;
-            emuenv.common_dialog.savedata.icon_loaded[idx] = true;
-        } else {
-            emuenv.common_dialog.savedata.icon_loaded[idx] = false;
         }
-    } else {
+    } else
         emuenv.common_dialog.savedata.title[idx] = emuenv.common_dialog.lang.save_data.save["new_saved_data"];
-        emuenv.common_dialog.savedata.icon_loaded[idx] = false;
-    }
 }
 
-static void check_save_file(SceUID fd, std::vector<SceAppUtilSaveDataSlotParam> slot_param, int index, EmuEnvState &emuenv, const char *export_name) {
-    vfs::FileBuffer thumbnail_buffer;
+static void check_save_file(const uint32_t index, EmuEnvState &emuenv, const char *export_name) {
+    SceUID fd = open_file(emuenv.io, construct_slotparam_path(emuenv.common_dialog.savedata.slot_id[index]).c_str(), SCE_O_RDONLY, emuenv.pref_path, export_name);
     if (fd < 0) {
         auto empty_param = emuenv.common_dialog.savedata.list_empty_param[index];
         check_empty_param(emuenv, empty_param, index);
     } else {
-        read_file(&slot_param[index], emuenv.io, fd, sizeof(SceAppUtilSaveDataSlotParam), export_name);
+        vfs::FileBuffer thumbnail_buffer;
+        SceAppUtilSaveDataSlotParam slot_param;
```

Comment:
Do initialize variable. If the file is empty then we have an undefined behavior.
```suggestion
        SceAppUtilSaveDataSlotParam slot_param{};
```